x-pack/plugins/security_solution/public/resolver/store/data/reducer.ts

```diff
@@ -36,6 +38,18 @@ export const dataReducer: Reducer<DataState, ResolverAction> = (state = initialS
       ...state,
       hasError: true,
     };
+  } else if (action.type === 'userRequestedRelatedEventData') {
+    state.relatedEventsReady.set(action.payload, false);
+    return {
+      ...state,
+      relatedEventsReady: new Map(state.relatedEventsReady.set(action.payload, false)),
+    };
+  } else if (action.type === 'serverReturnedRelatedEventData') {
+    return {
+      ...state,
+      relatedEventsReady: new Map(state.relatedEventsReady.set(action.payload.entityID, true)),
+      relatedEvents: new Map(state.relatedEvents.set(action.payload.entityID, action.payload)),
```

Comment:
Does the presence of `action.payload.entityID` in `relatedEvents` give enough information to determine if the related events is `ready`? Could we use a single map instead?

Reply:
That's a good question. The answer is "no" - at least the way the rest of the system is configured here, because you need to represent 3 things:
1) `not yet requested` : which the UI uses as a cue that it should request them
2) `requested but not yet received` : which the UI uses as a cue to stay in a waiting state
3) `received results`: for when you have them

You *could* multiplex them into one value, but I got absolutely pounded for that in my last PR review. Once bitten.